### PR TITLE
6 flight serialization

### PR DESCRIPTION
**Pull request by lucasiragusa**
Advance the development of the flight serialization logic. 

I am only left with two #TODOs plus the testing. 

I would like to merge all these pull requests because I am getting confused with all the branches

### CODE REVIEW

**Comment by lucasiragusa** [comment]
This is another thing I do not understand: there are currently two PRs displaying for branch 6-flight-serialization: 

From what I read, when you push new commits to a branch that has an open pull request (PR) on GitHub, the PR should automatically update to include those new commits.

This doesn't seem to be the case, so does it mean that we should use only this PR (which is the latest one) for this branch ?

**Comment by lucasiragusa** [comment]
As requested, I am adding a reminder that this is the PR linked to the issue #20 about slow performance. 

Some additional description of the issue: 
This issue can be observed by running either test_batch() or test_comparative() in the main. 

For test_batch, I was unable to wait until fill execution when calling 

```python main.py ../data/ssim/EY_SSIM.ssim```

Which is an airline schedule with ~5 years worth of flights for a large international airline. 

I created two smaller versions of this SSIM by cutting flights. A "Small" version, with about 20% of the original size, and a "Tiny" version with only one flight series. 

When executing 

```python main.py ../data/ssim/EY_SSIM_small_base.ssim ```

(venv) lucasiragusa@Lucas-MacBook-Pro src % python main.py ../data/ssim/EY_SSIM_small_base.ssim                                    
Creating permit for: AE
Time taken for AE: 321.51 seconds

Creating permit for: DE
Time taken for DE: 77.05 seconds

Creating permit for: IE
Time taken for IE: 3.55 seconds

Creating permit for: ES
Time taken for ES: 1.44 seconds

Creating permit for: GB
Time taken for GB: 9.57 seconds

Creating permit for: BE
Time taken for BE: 0.41 seconds

Creating permit for: DK
Time taken for DK: 0.29 seconds

Creating permit for: CH
Time taken for CH: 1.71 seconds

Creating permit for: FR
Time taken for FR: 1.70 seconds

Execution time: 427.25193309783936 seconds

When executing 

```python main.py ../data/ssim/EY_SSIM_tiny.ssim ```

(venv) lucasiragusa@Lucas-MacBook-Pro src % python main.py ../data/ssim/EY_SSIM_tiny.ssim      
Creating permit for: AE
Time taken for AE: 0.11 seconds

Creating permit for: DE
Time taken for DE: 0.12 seconds

Execution time: 0.24219322204589844 seconds

In summary, the "small" ssim is ~40 times larger than the "tiny" ssim, but it takes more than 1,700x longer to generate the permits for it. 

----

A similar problem occurs with test_comparative(). 

```venv) lucasiragusa@Lucas-MacBook-Pro src % python main.py ../data/ssim/EY_SSIM_small_base.ssim ../data/ssim/EY_SSIM_small_alt.ssim```

Countries with changes: ['AE', 'DE']
Creating permit for: AE
Time taken for AE: 313.70 seconds

Creating permit for: DE
Time taken for DE: 74.64 seconds

Execution time: 407.6537940502167 seconds

#TODO: Look at the code together and understand how to diagnose which part of the code is causing slow performance.

**Comment by Zaubeerer** [comment]
Hey @lucasiragusa, I just did a couple of things:
1. switched the target branch to `main`, because we already merged the streamlit frontend changes in a previous PR
2. I set `git config pull.rebase false`, so that we merge changes, if there are changes
3. I `pulled from main`, because this feature branch was not up-to-date with main. Make sure to pull from main every now and then and especially before creating a PR.

File: src/models/.DS_Store
Comment (Zaubeerer) [inline src/models/.DS_Store:1]: delete?

**Comment by Zaubeerer** [comment]
Hey @lucasiragusa, as this PR is already pretty large I am inclined to merge this branch ahead of the code clinic tomorrow in which Bob and I are going to create two feature branches and solve the imports with different approaches.

We can then create clearly-scoped follow-up issues and create smaller PRs for to solve each of the issues.

**Comment by Zaubeerer** [comment]
Let's merge this branch after the code clinic, to simplify comparing our PRs against the respective best target branch :)
